imap_processing/cli.py

```diff
@@ -574,39 +622,44 @@ def do_processing(self, dependencies: list) -> list[xr.Dataset]:
         print(f"Processing HIT {self.data_level}")
         datasets: list[xr.Dataset] = []
 
+        dependency_list = dependencies.processing_input
         if self.data_level == "l1a":
-            if len(dependencies) > 1:
+            if len(dependency_list) > 1:
                 raise ValueError(
                     f"Unexpected dependencies found for HIT L1A:"
-                    f"{dependencies}. Expected only one dependency."
+                    f"{dependency_list}. Expected only one dependency."
                 )
+            l0_file_obj = dependency_list[0].imap_file_paths[0]
             # process data to L1A products
-            datasets = hit_l1a(dependencies[0], self.version)
+            datasets = hit_l1a(l0_file_obj.construct_path(), self.version)
 
         elif self.data_level == "l1b":
-            if len(dependencies) > 1:
+            if len(dependency_list) > 1:
                 raise ValueError(
                     f"Unexpected dependencies found for HIT L1B:"
-                    f"{dependencies}. Expected only one dependency."
+                    f"{dependency_list}. Expected only one dependency."
                 )
             data_dict = {}
-            if self.dependencies[0]["data_level"] == "l0":
+            if dependency_list[0]["descriptor"] == "hk":
```

Comment:
```suggestion
            if dependency_list[0].descriptor == "hk":
```
dependency_list is a list of ProcessingInput objects, no longer a dictionary.

Reply:
Also, should this be .contains rather than ==?